lib/agent.js

```diff
@@ -22,6 +23,18 @@ function defaultFactory (origin, opts) {
     : new Pool(origin, opts)
 }
 
+function getDNSResolver (options) {
+  if (options?.dnsResolverOptions?.disable) {
```

Comment:
I wonder if it would be a better DX to have `{ dnsResolverOptions: false }` instead to completely disable it.
Not a blocker, just a thought.